projects/igniteui-angular/src/lib/core/styles/components/icon-button/_icon-button-theme.scss

```diff
@@ -0,0 +1,386 @@
+
+    @each $_name, $_schema in $icon-button-schema {
+        @if not($variant) {
+            $variant: map.get($schema, '_meta', 'theme');
+        }
+
+        @if not($foreground) and $background {
+            $foreground: text-contrast($background);
+        }
+
+        @if not($hover-background) and $background {
+            @if meta.type-of($background) == 'color' {
+                $hover-background: color.scale($background, $lightness: 5%);
+            }
+        }
+
+        @if not($hover-foreground) and $hover-background {
+            @if meta.type-of($hover-background) == 'color' {
+                $hover-foreground: text-contrast($hover-background);
+            }
+        }
+
+        @if not($focus-background) and $background {
+            $focus-background: color.scale($background, $lightness: 5%);
+        }
+
+        @if not($focus-foreground) and $focus-background {
+            $focus-foreground: text-contrast($focus-background);
+        }
+
+        $themes: map.merge($themes, (
+            $_name: extend( digest-schema($_schema), (
+                name: $name,
+                background: $background,
+                foreground: $foreground,
+
+                hover-background: $hover-background,
+                hover-foreground: $hover-foreground,
+
+                focus-background: $focus-background,
+                focus-foreground: $focus-foreground,
+
+                active-background: $active-background,
+                active-foreground: $active-foreground,
+
+                border-radius: $border-radius,
+                border-color: $border-color,
+                focus-border-color: $focus-border-color,
+
+                disabled-background: $disabled-background,
+                disabled-foreground: $disabled-foreground,
+                disabled-border-color: $disabled-border-color,
+                size: $size,
+            ))
+        ));
+    }
+
+    @return (
+        name: $name,
+        variant: $variant,
+        themes: $themes,
+    );
+}
+/// @param {Map} $theme - The theme used to style the component.
+@mixin icon-button($theme) {
+    $variant: map.get($theme, variant);
+    $flat-theme: map.get(map.get($theme, 'themes'), 'flat');
+    $outlined-theme: map.get(map.get($theme, 'themes'), 'outlined');
+    $contained-theme: map.get(map.get($theme, 'themes'), 'contained');
+
+    @include css-vars($flat-theme);
+    @include css-vars($outlined-theme);
+    @include css-vars($contained-theme);
+
+    $time: map.get((
+        material: .1s,
+        fluent: .1s,
+        bootstrap: .15s,
+        indigo-design: .15s
+    ), $variant);
+
+    $button-transition: color $time ease-in-out,
+        background-color $time ease-in-out,
+        border-color $time ease-in-out,
+        box-shadow $time ease-in-out;
+    
+    $icon-sizes: map.get((
+        material: rem(18px),
+        fluent: rem(16px),
+        bootstrap: rem(18px),
+        indigo-design: rem(18px),
+    ), $variant);
```

Comment:
Looking at the web components implementation, the icon should always be 18px in size, regardless of theme.